docs/discussion/selection-api.md

```diff
@@ -0,0 +1,326 @@
+*/
+function GetSelectedRegion()
+```
+
+Returns the region definition of the selection event, if a `Region` was used to select. This would require updates to **aladin-lite** to maintain knowledge of the regions used to select, which is not currently done explicitly, but can be accessed through the [selector Finite State Machines](https://github.com/cds-astro/aladin-lite/tree/master/src/js/FiniteStateMachine). 
+#### GetSelectedFootprints
+
+```
+/**
+*@returns List[Footprint]
+*/
+function GetSelectedFootprints()
+```
+
+Returns the currently selected `List[Footprint]`
+## 3.3. SELECT
+#### SelectSources 
+##### Option 1: 
+
+```
+/**
+* @param sources: List[Source]
+* @param match_keys: List[str]
+*/
+function SelectSources(sources, match_keys)
+```
+
+Selects the list of sources provided by the user within **aladin-lite**. The list of keys provided in `match_keys` are used to match the sources with corresponding entries in the catalogs.
+
+| **Parameter** | **Type**     | **Description**                              |
+| ------------- | ------------ | -------------------------------------------- |
+| sources       | List[Source] | The list of sources to select.               |
+| match_keys    | List[str]    | The list of keys to use to determine a match |
+
+-----------------------------------------------------------------------------
+##### Option 2:
+
+```
+/**
+* @param match_func: func()
+*/
+function SelectSources(match_func)
+```
+
+Selects the list of sources provided. Uses the `match_func` provided to determine if a source should be selected. This is similar to how region selections are currently performed in **aladin-lite**, where a function [s.contains( ... )](https://github.com/cds-astro/aladin-lite/blob/master/src/js/FiniteStateMachine/CircleSelect.js#L85-L102) is defined. 
+
+| **Parameter** | **Type**  | **Description**                                                |
+| ------------- | --------- | -------------------------------------------------------------- |
+| match_func    | func(a,b) | The function used to determine if an object should be selected |
+
+-------------------------------------------------------------------------------
+#### SelectRegion
+
+```
+/**
+* @param region: Region
+*/
+function SelectRegion(region)
+```
+
+Triggers a `Region` selection within **aladin-lite** widget. This function would need to mock of the various state transitions expected by the FiniteStateMachines used for region selection ([example](https://github.com/cds-astro/aladin-lite/blob/master/src/js/FiniteStateMachine/CircleSelect.js#L137-L165)).
+
+| **Parameter** | **Type** | **Description**                     |
+| ------------- | -------- | ----------------------------------- |
+| region        | Region   | The region to select objects within |
+
+-----------------------------------------------------------------------
+#### SelectFootprints
+
+```
+/**
+* @param footprints: List[Footprint]
+*/
+function SelectFootprints(footprints)
```

Comment:
The question above in the python API about how to identify footprints probably applies here too.